node/rpc/src/chain_api.rs

```diff
@@ -109,24 +109,13 @@ where
             .await
     } {
         Ok(checksum) if !dry_run => {
-            std::fs::rename(&out_tmp, &output_path)?;
```

Comment:
I believe there's similar logic in `snapshot fetch`, would u create a tracking issue if not include that in this PR?

Reply:
yes I'll create a new ticket for handling similar cases for `snapshot fetch`